### ISSUE TITLE
client inside a future inside a System::block_on inside a web::block inside a request processing

### ISSUE THREAD

**Issue by unegare** [actix/actix-web#1007]
how to replay:

[link to repo](https://github.com/unegare/rust-actix-rest/tree/actix_bug_found)

run `cargo run` and `tests/actix_crash.sh`

After a few seconds it causes the error:
```
Connect(Resolver(ResolveError { inner: 

background resolver gone, this is a bug }))
```

**Comment by jesskfullwood** [comment]
Reduced:
```rust
use actix_web::{web, client::Client, App, HttpServer};

use futures::Future;

fn bug() -> Box<dyn Future<Item = (), Error = ()>> {
    Box::new(
        actix_web::web::block(|| {
            actix_rt::System::new("fut").block_on(futures::lazy(|| {
                let client = Client::new();
                client.get("http://www.google.com")
                    .send()
                    .map(|_| ())
                    .map_err(|e| {
                        println!("client err, {:?}", e);
                    })
            }))
        }).map_err(|_| ())
    )
}

fn main() {
    HttpServer::new(move || {
        App::new()
            .route("/", web::get().to(bug))
    })
    .bind("0.0.0.0:8080").unwrap()
    .run().unwrap();
}
```
Test with
```
seq 50 | xargs -i curl localhost:8080/ &
```

**Comment by fafhrd91** [comment]
this requires arbiter specific data storage

**Comment by fafhrd91** [comment]
fixed in actix-connect 0.2.4
